conda_anaconda_tos/api.py

```diff
@@ -325,3 +334,251 @@ def clean_tos(tos_root: str | os.PathLike[str] | Path) -> Iterator[Path]:
+                continue
+
+    # Create metadata file
+    metadata = {
+        "timestamp": backup_timestamp.isoformat(),
+        "backup_type": "comprehensive",
+        "files_count": len(file_info),
+        "source_locations": list({info.source_location for info in file_info}),
+        "temporary_locations": list(
+            {info.source_location for info in file_info if info.is_temporary}
+        ),
+        "persistent_locations": list(
+            {info.source_location for info in file_info if not info.is_temporary}
+        ),
+        "files": [
+            {
+                "path": str(info.file_path),
+                "source_location": info.source_location,
+                "is_temporary": info.is_temporary,
+            }
+            for info in file_info
+        ],
+        "version": APP_VERSION,
+    }
+
+    metadata_file = timestamped_backup_dir / "metadata.json"
+    try:
+        with metadata_file.open("w") as f:
+            json.dump(metadata, f, indent=2)
+    except (PermissionError, OSError) as e:
+        raise CondaToSBackupError("write metadata", metadata_file, str(e)) from e
+
+    return timestamped_backup_dir
+
+
+def restore_tos_configs(
+    timeout_hours: float = 1.0,
+) -> tuple[bool, list[Path]]:
+    """Restore ToS configuration files from recent backup to exact original locations.
+
+    Args:
+        timeout_hours: Timeout in hours for considering backups recent
+
+    Returns:
+        - bool: True if any restore was performed, False otherwise
+        - list[Path]: List of successfully restored files
+
+    Raises:
+        FileNotFoundError: If the backup directory does not exist
+
+    """
+    backup_dir = TOS_BACKUP_DIR
+
+    if not backup_dir.exists():
+        raise FileNotFoundError(f"Backup directory does not exist: {backup_dir}")
+
+    # Find recent backups within timeout period
+    timeout_cutoff = datetime.now(tz=timezone.utc) - timedelta(hours=timeout_hours)
+    recent_backups = []
+
+    for individual_backup_dir in backup_dir.iterdir():
+        if (
+            not individual_backup_dir.is_dir()
+            or not individual_backup_dir.name.startswith("backup_")
+        ):
+            continue
+
+        metadata_file = individual_backup_dir / "metadata.json"
+        if not metadata_file.exists():
+            continue
+
+        try:
+            with metadata_file.open() as f:
+                metadata = json.load(f)
+
+            backup_time = datetime.fromisoformat(metadata["timestamp"])
+            if backup_time >= timeout_cutoff:
+                recent_backups.append((backup_time, individual_backup_dir, metadata))
+        except (json.JSONDecodeError, KeyError, ValueError):
+            # Invalid metadata, skip this backup
+            continue
+
+    if not recent_backups:
+        return False, []
+
+    # Use the most recent backup
+    _, latest_backup_dir, metadata = max(recent_backups, key=lambda x: x[0])
```

Comment:
If we always get the latest backup, do we need timestamped instead of rolling backups? As far as I can see, the timestamp is only used for a cut-off time, which can also be stored in the metadata.

Or are there use cases to merge different backups in the future?